apps/verticalhome/js/app.js

```diff
@@ -150,6 +164,206 @@
+
+              obj.icon = typeof(
+                app.pinnedAppsManager.items[i].entry.icon
+                )=='undefined'?
+                app.pinnedAppsManager.items[i].entry.icon.baseURI
+                :app.pinnedAppsManager.items[i].entry.icon;
+
+              obj.index = app.pinnedAppsManager.items[i].index;
+
+              app.itemStore.savePinnedAppItem(obj);
+
+          }
+        }
+
+        var notif = document.getElementById('blinknotif');
+        notif.style.display = 'block';
+        notif.innerHTML = 'Changes saved';
+
+        setTimeout(function() {
+          notif.style.display = 'none';
+        }, 3000);
+
+    },
+
+    pin2MainScreen: function(){
+      var selectedMoreAppItem = document
+        .querySelector('#more-apps-screen .selected');
+
+      var selectedMoreAppItem_icon = selectedMoreAppItem
+        .getAttribute('data-test-icon');
+
+      var selectedMoreAppItem_data_identifier = selectedMoreAppItem
+        .getAttribute('data-identifier');
+
+      var selectedMoreAppItem_title = selectedMoreAppItem
+        .querySelector('span.title').innerHTML;
+
+      var notif = document.getElementById('blinknotif');
+      var maxDataIndex = app.pinnedAppsManager.items.length;
+
+      app.pinnedAppsManager.addEntry( maxDataIndex,
+                                      selectedMoreAppItem_data_identifier,
+                                      selectedMoreAppItem_icon,
+                                      selectedMoreAppItem_title);
+
+      Array.prototype.slice.call(
+        document.querySelector('#more-apps-screen #icons').children
+      );
+
+      app.itemStore.applicationSource.
+        removeIconFromGrid( selectedMoreAppItem_data_identifier );
+
+      this.renderGrid();
+
+      if (selectedMoreAppItem.parentNode) {
+        selectedMoreAppItem.parentNode.removeChild(selectedMoreAppItem);
+      }
+
+      this.hideMoreApps();
+
+      notif.style.display = 'block';
+      notif.innerHTML = 'App added to Home Screen';
+
+      setTimeout(function() {
+        notif.style.display = 'none';
+      }, 3000);
+
+      var lastTimeSelected = app.pinnedAppsNavigation.currentIndex();
+      var insrertedIndex = app.pinnedAppsManager.items.length;
+      var scrollDiff = insrertedIndex - maxDataIndex;
+      app.pinnedAppsNavigation.elements[lastTimeSelected].classList
+        .remove('selected');
+
+      for(var z = 0; z < scrollDiff; z++){
+        var itm = app.pinnedAppsNavigation.elements[0];
+        var cln_tmp = itm.cloneNode(true);
+        app.pinnedAppsNavigation.elemList.appendChild(cln_tmp);
+        itm.parentNode.removeChild(itm);
+      }
+
+      app.pinnedAppsNavigation.elements[1].classList.add('selected');
+
+      app.pinnedAppsNavigation.scrollNum();
+
+      app.pinnedAppsNavigation.refresh();
+
+    },
+
+    pin2MoreApp: function(){
+
+      var selectedPinAppItem = document.querySelector(
+        '#pinned-apps-list .selected'
+        );
```

Comment:
It is used in many places. 
I suggest to save this as class member.
this.selectedSelector = '#pinned-apps-list .selected'